api/controllers/StatsController.js

```diff
@@ -96,40 +131,61 @@ function transformSeasonToDTO(season) {
   return {
     ...rest,
     rankings: rankingsAsArray,
+    uniquePlayersPerWeek: season.uniquePlayersPerWeek.map((playerSet) => playerSet.size),
   };
 }
 
+
 module.exports = {
-  getStats: function (req, res) {
+  getStats: function (_req, res) {
     // Find records
     const seasons = sails.helpers.getSeasonsWithoutRankings();
     const matches = Match.find({});
     const users = User.find({});
-    return Promise.all([seasons, matches, users]).then(([seasons, matches, users]) => {
+    const games = Game.find({ result: {'>=': -1} }); // Only find completed games
+    return Promise.all([seasons, matches, users, games]).then(([seasons, matches, users, games]) => {
+
       const idToUserMap = new Map();
       users.forEach((user) => {
         idToUserMap.set(user.id, user);
       });
+
       // Add each match to the appropriate rankings
       matches.forEach((match) => {
         // Only count finished matches
         if (match.endTime && match.winner) {
-          const relevantSeason = seasons.find((season) => {
-            // Find season this match took place during
-            return dayjs(match.startTime).isBetween(dayjs(season.startTime), dayjs(season.endTime));
-          });
-          if (relevantSeason) {
-            const player1 = idToUserMap.get(match.player1);
-            const player2 = idToUserMap.get(match.player2);
-            if (player1 && player2) {
-              // Player 1
-              addMatchToRankings(relevantSeason, match, player1, player2);
-              // Player 2
-              addMatchToRankings(relevantSeason, match, player2, player1);
-            }
+          const relevantSeason = getSeasonFromTimeStamp(seasons, match.startTime);
+          if (!relevantSeason) {
+            return;
+          }
+          const player1 = idToUserMap.get(match.player1);
+          const player2 = idToUserMap.get(match.player2);
+          if (player1 && player2) {
+            // Player 1
+            addMatchToRankings(relevantSeason, match, player1, player2);
+            // Player 2
+            addMatchToRankings(relevantSeason, match, player2, player1);
           }
         }
       });
+
+      games.forEach((game) => {
+        const relevantSeason = getSeasonFromTimeStamp(seasons, game.updatedAt);
+        if (!relevantSeason) {
+          return;
+        }
+
+        const weekNum = dayjs(game.updatedAt).diff(relevantSeason.startTime, 'week');
+        if (!relevantSeason.gameCounts[weekNum]) {
+          relevantSeason.gameCounts[weekNum] = 0;
+        }
+        relevantSeason.gameCounts[weekNum]++;
+        if (!relevantSeason.uniquePlayersPerWeek[weekNum]) {
+          relevantSeason.uniquePlayersPerWeek[weekNum] = new Set();
+        }
```

Comment:
@seriouslysean and @byarmis these nullish checks and re-initialization logic are the key to fixing the null pointer exception that crashed prod when I merged #469. Basically the Spades 2022 season started at 8:30pm on a Wednesday and ended at 6:00pm on a Wednesday 11 weeks later, which means the difference in weeks betwen start and end time as computed by `dayjs` is actually 10, even though it is only 2.5 hours short of being 11 weeks long. So any game in that 11th week would cause an NPE when trying to add the player ids to the set for that week on the lines below, which crashed the server. 

The `gameCounts` calculation was also wrong b/c it would get NaN and keep trying to increment and failing, but the `.add()` below is what actually crashed the server.

I considered trying to update the initialization of the seasons in `get-seasons-without-rankings` to somehow round to accommodate this, but I think it's actually more defensive and type safe to just make sure that any value in the array has a value before we update it here.

Reply:
Actually maybe it does make more sense to just update the initialization in the helper. Per the `dayjs` [docs](https://day.js.org/docs/en/display/difference) we can make diff output a float instead of truncating, which would let us round up. That should solve the problem too, maybe more elegantly